### PR TITLE
use enumPropertyFactory validator to check values

### PR DESCRIPTION
**Pull request by soleneG02**

### CODE REVIEW

File: src/elements/GeoDropdown/GeoDropdown.vue [hunk 1]
```diff
@@ -117,9 +105,10 @@ export default {
...
       validator: function (value) {
-        if (value === undefined || value in GeoDropdownConstants.Y_AXIS_POSITION) return true
+        const allValues = Object.values(GeoDropdownConstants.Y_AXIS_POSITION)
+        if (value === undefined || allValues.includes(value)) return true
 
-        const supportedValues = Object.values(GeoDropdownConstants.Y_AXIS_POSITION).map(i => `«${i}»`).join(', ')
+        const supportedValues = allValues.map(i => `«${i}»`).join(', ')
         console.warn(`GeoDropdown [component] :: Unsupported value («${value}») for «forceYAxisPosition» property. Use one of ${supportedValues}`)
         return false
       }
```
Comment (riboher) [inline src/elements/GeoDropdown/GeoDropdown.vue]: Why not using the enumPropertyFactory here?

File: src/elements/GeoInput/GeoInputMessage.vue [hunk 2]
```diff
@@ -43,10 +43,10 @@ export default {
...
-        if (!value) return true
-        if (value in VARIANTS) return true
+        const allValues = Object.values(VARIANTS)
+        if (!value || allValues.includes(value)) return true
 
-        const supportedValues = Object.values(VARIANTS).map(i => `«${i}»`).join(', ')
+        const supportedValues = allValues.map(i => `«${i}»`).join(', ')
         console.warn(`GeoInputMessage [component] :: Unsupported value («${value}») for «variant» property. Use one of ${supportedValues}`)
         return false
       }
```
Comment (riboher) [inline src/elements/GeoInput/GeoInputMessage.vue]: Same, can't you use it here as well?

File: src/elements/GeoSelect/GeoSelect.mixin.js [hunk 3]
```diff
@@ -208,7 +209,12 @@ export default {
...
       required: false,
       validator: function (value) {
-        return value === undefined || value in Y_AXIS_POSITION
+        const allValues = Object.values(Y_AXIS_POSITION)
+        if (value === undefined || allValues.includes(value)) return true
+
+        const supportedValues = allValues.map(i => `«${i}»`).join(', ')
+        console.warn(`GeoSelect [component] :: Unsupported value («${value}») for «forceYAxisPosition» property. Use one of ${supportedValues}`)
+        return false
       }
```
Comment (riboher) [inline src/elements/GeoSelect/GeoSelect.mixin.js]: Same. Use it here?

File: src/elements/GeoDropdown/GeoDropdown.vue (hunk 1 above)
Comment (fofi) [reply src/elements/GeoDropdown/GeoDropdown.vue]: I think because here we are allowing undefined

File: src/utils/enumPropertyFactory.js [hunk 4]
```diff
@@ -1,18 +1,46 @@
...
+ * @param {object} params
+ * @param {Boolean} [params.required]
+ * @param {Object<string, string>} params.enumDictionary
+ * @param {string} params.componentName
+ * @param {string} params.propertyName
+ */
+function enumRequiredPropertyFactory (params) {
+  return {
+    type: String,
+    required: params.required ? params.required : false,
```
Comment (fofi) [inline src/utils/enumPropertyFactory.js]: We can simple cast the value to boolean using `!!params.required`

File: src/utils/enumPropertyFactory.js [hunk 5]
```diff
@@ -1,18 +1,46 @@
...
+
 /**
  * @param {object} params
  * @param {string} [params.defaultValue]
  * @param {Object<string, string>} params.enumDictionary
  * @param {string} params.componentName
  * @param {string} params.propertyName
  */
-export default function enumPropertyFactory (params) {
+function enumDefaultPropertyFactory (params) {
```
Comment (fofi) [inline src/utils/enumPropertyFactory.js]: Maybe we can use the same function, and depending on the parameter add one property or other.
For example
```
const defaultEnumProperty = {
  type: String,
  validator
  ....
}
if ('defaultValue' in params) {
  defaultEnumProperty.default = params.defaultValue
}
if (required...
```

Comment (fofi) [reply src/utils/enumPropertyFactory.js]: And maybe we could also add a property in params to check `value === undefined`.

We can discuss it later on which approach take

**Comment by fofi** [comment]
Nice work on the refactor :recycle:

File: src/elements/GeoInput/GeoInputMessage.vue (hunk 2 above)
Comment (soleneG02) [reply src/elements/GeoInput/GeoInputMessage.vue]: It was because of the !value as well, but I introduced it in the function

**Pull request merged**